### PR TITLE
tooling: add formatter, upgrade eslint and create pre-commit hook

### LINKED ISSUE #141: Introduce a formatter

### PR DESCRIPTION
**Pull request by stickyPiston**
So as to keep the code formatting consistent across the project, this PR proposes to add prettier (for formatting) and upgrade eslint (for linting). Furthermore, this PR includes the setup for husky (which manages a new pre-commit hook) and lint-staged (which applies prettier and eslint on staged files and auto-stages the changes in the commit).

Closes #141

### CODE REVIEW

File: .eslintrc.json
Comment (SilasPeters) [inline .eslintrc.json:1]: I spent so much time on this :(

**Comment by stickyPiston** [comment]
@josvanos I've implemented all of your feedback on all of the PRs. Would you mind reviewing this PR - which has all of the changes from all PRs merged into it - once again?

**Comment by josvanos** [comment]
Code wise it looks great, I think it would be beneficial if this would be merged in the codebase soon.

File: src/components/Activities.tsx [hunk 1]
```diff
@@ -22,13 +21,13 @@ export default function Activities({ current }: StateMachineSlideProps) {
   return (
     <div>
       <ul className="activities">
-        {activities.map((activity, i) =>
+        {activities.map((activity, i) => (
```
[the remark this answers is not in the document]
Comment (stickyPiston) [reply src/components/Activities.tsx]: Good catch, i've immediately changed it!

**Comment by stickyPiston** [comment]
@josvanos Thanks heaps for your time once again!

**Pull request merged**